install_config/configuring_vsphere.adoc

```diff
@@ -0,0 +1,236 @@
+|Cluster,
+Hosts,
+VM Folder
+|Yes
+
+|manage-k8s-volumes
+|Datastore.AllocateSpace
+Datastore.FileManagement
+System.Anonymous
+System.Read
+System.View
+|Datastore
+|No
+
+|k8s-system-read-and-spbm-profile-view
+|StorageProfile.View
+System.Anonymous
+System.Read
+System.View
+|vCenter
+|No
+
+|ReadOnly
+|System.Anonymous
+System.Read
+System.View
+|Datacenter,
+Datastore Cluster,
+Datastore Storage Folder
+|No
+
+|===
+
+. Create the xref:../../install_config/persistence_storage/persistence_storage_vsphere.adoc#vsphere-configuration-file[vSphere cloud config file `vsphere.conf`] and place it in the shared directory which is accessible from kubelet container, controller-manager pod, and the API server pod.
+
+.  Add following flags to kubelet running on every node and to the controller-manager and API server pods manifest files:
++
+[source,bash]
+----
+--cloud-provider=vsphere
+--cloud-config=<Path of the vsphere.conf file>
+----
+
+. Restart Kubelet on all nodes. To do this, reload kubelet systemd unit file using `systemctl daemon-reload` and then restart kubelet service using the command `systemctl restart kubelet.service`.
++
+[NOTE]
+====
+After enabling the vSphere Cloud Provider, Node names are set to the VM names from the vCenter Inventory.
+====
+
+[[vsphere-configuration-file]]
+== The VMWare vSphere Configuration File
+Configuring {product-title} for VMWare vSphere requires the *_/etc/vsphere/vsphere.conf_* file, on each node host.
+
+If the file does not exist, create it, and add the following:
+
+----
+[Global]
+        user = "username" <1>
+        password = "password" <2>
+        server = "10.10.0.2" <3>
+        port = "443" <4>
+        insecure-flag = "1" <5>
+        datacenter = "datacenter-name" <6>
+        datastore = "datastore-name" <7>
+        working-dir = "vm-folder-path" <8>
+        vm-name = "vm-name-of-master-node" <9>
+        vm-uuid = "vm-uuid" <10>
+[Disk]
+    scsicontrollertype = pvscsi
+----
+<1> vCenter username for the vSphere cloud provider.
+<2> vCenter password for the spcified user.
+<3> IP Address or FQDN for the vCenter server.
+<4> (Optional) Port number for the vCenter server. (Defaults to port `443`)
+<5> Set to 1 if the vCenter uses a self-signed cert.
+<6> Name of the data center on which Node VMs are deployed.
+<7> Name of the datastore to use for provisioning volumes using the storage classes or dynamic provisioning. If datastore is located in storage folder or datastore is member of datastore cluster, specify the full datastore path. Verify that vSphere Cloud Provider user has Read Privilege set on the datastore cluster or storage folder to be able to find datastore.
+<8> The vCenter VM folder path, in which the node VMs are located.  It can be set to an empty path(`working-dir = ""`), if Node VMs are located in the root VM folder.
+<9> (Optional) VM name of the Master Node. If this parameter is specified , `vsphere.conf` file on the worker node does not need vCenter credentials.
```

Comment:
According to https://kubernetes.io/docs/getting-started-guides/vsphere/, `vm-name` is introduced in kubernetes 1.6.4. Our latest OCP has kubernetes `v1.6.1+5115d708d7`. So I think this parameter isn't supported in our product yet.